### PR TITLE
FEAT - implement game model, collection, controller and tests

### PR DESCRIPTION
**Pull request by travisCarson**
## Description

Create game model and methods
Create game collection and methods
Create game controller
Create game settings
Create game tests
## Related Issue(s)

closes smileguess/smileguess #36
## How Has This Been Tested?

Created new tests with Jasmine
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Major refactor (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (Check this if you changed any documentation at all)
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have rebased and squashed my commits.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: models/Game.js [hunk 1]
```diff
@@ -19,10 +28,75 @@
...
 module.exports = class Game {
   constructor(userWhoStartsGame) {
+    this.gameId = null;
+    this.socketId = null;
     this.players = [userWhoStartsGame];
+    this.seatsOpen = settings.maxPlayers - 1;
     this.dealer = null;
-    this.solution = null;
+    this.solutionForDisplay = null;
+    this.solutionForMatcing = null;
```
Comment (willfulbard) [inline models/Game.js]: Change to matching

File: models/Game.js [hunk 2]
```diff
@@ -19,10 +28,75 @@
 
 module.exports = class Game {
   constructor(userWhoStartsGame) {
+    this.gameId = null;
+    this.socketId = null;
     this.players = [userWhoStartsGame];
+    this.seatsOpen = settings.maxPlayers - 1;
     this.dealer = null;
-    this.solution = null;
+    this.solutionForDisplay = null;
```
Comment (willfulbard) [inline models/Game.js:66]: Please add comment about the functionality of this, it's ambiguous.

File: models/Game.js [hunk 3]
```diff
@@ -19,10 +28,75 @@
...
+    this.solutionForMatcing = null;
     this.category = null;
     this.clue = [];
```
Comment (willfulbard) [inline models/Game.js]: Why is this an array, should it actually be "clues" plural instead of "clue" singular?

File: models/Game.js (hunk 1 above)
Comment (travisCarson) [inline models/Game.js]: dunno how you spotted that.. but.. fixed!

File: models/Game.js (hunk 2 above)
Comment (travisCarson) [inline models/Game.js:66]: done

File: models/Game.js (hunk 3 above)
Comment (travisCarson) [inline models/Game.js]: I'm envisioning the clue as an array of emojis

File: models/Game.js [hunk 4]
```diff
@@ -19,10 +28,75 @@
...
+    this.active = false;
+  }
+
+  setOpenSeats() {
```
Comment (willfulbard) [inline models/Game.js]: the "set" word implies that I'm feeding it a value, like `this.setOpenSeats(5)`, perhaps rename to `updateSeatCount`

File: models/Game.js [hunk 5]
```diff
@@ -19,10 +28,75 @@
...
+    this.setOpenSeats();
+    this.classifyGame(this);
+    if (!this.active && this.players.length >= settings.minPlayers) {
+      this.active = true;
+      this.assignDealer();
+      this.getSolution();
+    }
+  }
+
+  classifyGame() {
```
Comment (willfulbard) [inline models/Game.js]: consider renaming to `updateGameAvailability`

File: models/Game.js (hunk 3 above)
Comment (travisCarson) [inline models/Game.js]: we have chosen to remove this property for the time being, substituting a comment in its place

File: models/Game.js (hunk 4 above)
Comment (travisCarson) [inline models/Game.js]: done

File: models/Game.js (hunk 5 above)
Comment (travisCarson) [inline models/Game.js]: done

**Pull request merged**